_ap/18/index.adoc

```diff
@@ -3,76 +3,200 @@ num: 18
+
+Most of our toolchains are (in some ways) using `openapi-generator`, developed and maintained by the "OpenAPI Tools" organization (which is NOT affiliated with the OpenAPI initiative).
+
+Unfortunately, OpenAPI Generator is not a well maintained OSS project:
+image::openapi-generator-gh.png[]
+
+Even our own attempts to fix known bugs have been ineffective:
+https://github.com/OpenAPITools/openapi-generator/pull/13616
+
+It's also pretty common to have breaking changes in the generator tooling, which results in breaking changes in the generated code.
+When this happens it can be hard to detect and often happens without a clear migration path.
+
+For these reasons we are often https://github.com/redhat-developer/app-services-sdk-go/blob/13d1cad9a937c456a9623ad188c44790e3370d83/openapitools.json#L5[stuck on old versions], preventing (in some cases) the opportunity to fix CVEs and issues by keeping transitive dependencies and used frameworks updated.
+
+Following are a variety of critical/important issues we currently have due to our OpenAPI Generator based approach to API Client generation.
+
+=== Security
+
+We should be able to keep our dependency trees under control and update/refresh whenever a vulnerability in the stack is detected.
+
+=== Technical debt
+
+We want to be able to improve the quality of the generated clients and SDKs and improve them over time, introducing even structural bug fixes and improvements.  Overall we strive for our code to be well structured and maintainable, even when that code is being generated (to the best extent possible).
+
+=== Community
+
+We want our community of customers to be able to use best-in-class tools to perform their day-to-day work. OpenAPI generation is an important piece of tooling and we should be able to provide a much better OSS experience than the one we are right now.
+
+== Goals
+
+* Identify alternative tooling that can be used in place of the current OpenAPI generator
+* Validate and improve the integration patterns and the user experience
+* Use up-to-date tooling and libraries for our customer-facing API clients
+
+== Non-goals
+
+* Dictate a technical decision for internal usage or for projects that are successfully rolling their own SDKs and clients (e.g. by manually curating them)
+
+== Current Architecture
+
+Inside Red Hat we are mostly relying (in various ways) on OpenAPI Generator to generate clients, we also maintain a https://github.com/quarkiverse/quarkus-openapi-generator[Quarkus Extension] based on the same generator.
+
+== Proposed Architecture
+
+We propose to switch to alternative OSS tooling provided by Microsoft called https://github.com/microsoft/kiota[Kiota].
+
+=== Advantages:
+
+* Community
+* Better design/architecture
+* Better developer experience
+* Maintenance
+* Swappable implementation libraries
+* Compatibility matrix
+
+=== Disadvantages:
+
+* Not GA (yet)
+* External Owners (Microsoft)
+* Core Generator codebase is .NET
```

Comment:
It is not a key aspect for this ADR but does it imply that this library cannot be integrated with Quarkus to replace the existing Quarkus OpenAPI extension or in general any possible Quarkus similar use cases?

Reply:
Let me try to provide more context here:

- `this library cannot be integrated with Quarkus` -> Kiota ships as a standalone CLI + Java dependencies and will work with Quarkus (possibly it will need a little bit of tweaking for `native-image`, but it should be it). We are already publishing a Maven plugin to easily integrate with Java workflows: https://github.com/redhat-developer/kiota-java-extra#maven-plugin
- `replace the existing Quarkus OpenAPI extension` -> replacing the current extension is out of the scope of this AP and out of the scope of Kiota. The Quarkus extension is vertical and highly integrated with an opinionated language and framework, while Kiota privileges a more consistent approach across different languages and decouples over the used frameworks. I plan to agree on an action plan with Ricardo to see if it makes sense to support both approaches, but no strong commitment here.
- `or in general any possible Quarkus similar use cases?` -> can you, please, elaborate if there is any aspect that I might have missed with the previous answers?

Reply:
Thanks for the clarification, you covered all my questions :)